test/cpp/interop/stress_test.cc

```diff
@@ -0,0 +1,221 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *is % allowed in string
+ */
+
+#include <memory>
+#include <string>
+#include <thread>
+#include <utility>
+#include <vector>
+
+#include <gflags/gflags.h>
+#include <grpc/support/time.h>
+#include <grpc++/create_channel.h>
+#include <grpc++/grpc++.h>
+
+#include "test/cpp/interop/interop_client.h"
+#include "test/cpp/interop/stress_interop_client.h"
+#include "test/cpp/util/test_config.h"
+
+DEFINE_int32(sleep_duration_ms, 0,
+             "The duration (in millisec) between two"
+             " consecutive test calls (per server) issued by the server.");
+
+DEFINE_int32(test_duration_secs, -1,
+             "The length of time (in seconds) to run"
+             " the test. Enter -1 if the test should run continuously until"
+             " forcefully terminated.");
+
+DEFINE_string(server_addresses, "localhost:8080",
+              "The list of server"
+              " addresses in the format:\n"
+              " \"<name_1>:<port_1>,<name_2>:<port_1>...<name_N>:<port_N>\"\n"
+              " Note: <name> can be servername or IP address.");
+
+// TODO(sreek): Add more test cases here in future
+DEFINE_string(test_cases, "",
+              "List of test cases to call along with the"
+              " relative weights in the following format:\n"
+              " \"<testcase_1:w_1>,<testcase_2:w_2>...<testcase_n:w_n>\"\n"
+              " The following testcases are currently supported:\n"
+              "   empty_unary\n"
+              "   large_unary\n"
+              "   large_compressed_unary\n"
+              "   client_streaming\n"
+              "   server_streaming\n"
+              "   empty_stream\n"
+              " Example: \"empty_unary:20,large_unary:10,empty_stream:70\"\n"
+              " The above will execute 'empty_unary', 20% of the time,"
+              " 'large_unary', 10% of the time and 'empty_stream' the remaining"
+              " 70% of the time");
+
+using std::make_pair;
+using std::pair;
+using std::string;
+using std::thread;
+using std::vector;
+
+using grpc::testing::kTestCaseList;
+using grpc::testing::StressTestInteropClient;
+using grpc::testing::TestCaseType;
+using grpc::testing::TestSelector;
+using grpc::testing::UNKNOWN_TEST;
+
+TestCaseType get_test_type_from_name(const string& test_name) {
```

Comment:
Use camelcase as per the style guide: https://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Function_Names